src/cosalib/ostree.py

```diff
@@ -0,0 +1,221 @@
+                self._compose_data = json.load(data)
+            log.info(f"Using OSTree ref '{self.ref_name}'")
+        finally:
+            os.chdir(build.workdir)
+
+    @property
+    def build(self):
+        return self._build
+
+    @property
+    def ostree_repo(self):
+        return self._ostree_repo
+
+    @property
+    def ref_name(self):
+        return self._compose_data.get("ref",
+                                      f"tmpref-{self.build.build_name}")
+
+    @property
+    def ostree_repo_size(self):
+        ors = getattr(self, "_ostree_repo_size", None)
+        if ors is None:
+            self._ostree_repo_size = estimate_tree_size(
+                self.ostree_repo, self.ref_name)
+        return self._ostree_repo_size
+
+    @property
+    def ostree_repo_estimated_size(self):
+        return self.ostree_repo_size.get("estimated-mb", {}).get("final", 4096)
+
+    @property
+    def ostree_tarball_meta(self):
+        return self.meta.get("images", {}).get("ostree", {})
+
+    @property
+    def ostree_tarball_path(self):
+        return self.ostree_meta.get("path")
+
+    @property
+    def ostree_tarball_sha256(self):
+        return self.ostree_meta.get("sha256")
+
+    @property
+    def ostree_size(self):
+        return self.ostree_meta.get("size")
+
+    def need_tree(self):
+        log.info("Checking for OSTree")
+        repo_path = self.ostree_repo
+        create = False
+        try:
+            _ = self.ostree_repo
+        except OSTreeException:
+            log.debugf(f"OSTree is missing, creating new one at {repo_path}")
+            create = True
+
+        if not create:
+            return
+
+        if self.ostree_path is None:
+            raise OSTreeException(f"no ostree in meta.json")
+
+        with tarfile.open(self.ostree_path, 'r:*') as tf:
+            log.info(f"extracting {self.ostree_path} to {self.ostree_repo}")
+            tf.extractall(path=self.ostree_repo, numeric_owner=True)
+
+    def create_ref(self):
+        ref = getattr(self, "_ostree_ref", self.ref_name)
+        if self.meta.get('ostree-commit'):
+            log.debug("ostree has been commited, skipping ref creation")
+            return
+
+        log.debug("found temp ostree ref, creating commit")
+        # create the ref
+        run_verbose([
+            "ostree", "refs",
+            f"--repo={self.ostree_repo}",
+            self.ostree_commit,
+            "--create", ref
+        ])
+        new_ref = self.get_rev(use_ref=ref)
+        log.info(f"mapped '{ref}' to '{new_ref}'")
+        self.meta['ostree-commit'] = new_ref
+        self.meta.write()
+
+    def get_rev(self, use_ref=None):
+        """
+        Get the OSTree ref based on the name.
+        """
+        cmd = [
+            "ostree", "rev-parse",
+            f"--repo={self.ostree_repo}",
+            use_ref
+        ]
+        rev = run_verbose(cmd, capture_output=True, check=True).stdout
+        return rev.decode("utf-8").strip()
```

Comment:
It's more efficient to pass `encoding='utf-8'` to kwargs and let the lib decode.